### PR TITLE
NAS-132595 / 25.04 / Handle non-JSON progress messages in TrueCloud Backup

### PR DESCRIPTION
**Pull request by creatorcary**
Even with the `--json` flag, not all restic messages are reported in JSON format.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/cloud_backup/restic.py [hunk 1]
```diff
@@ -86,8 +86,15 @@ async def restic_check_progress(job, proc, track_progress=False):
         read = (await proc.stdout.readline()).decode("utf-8", "ignore")
         if read == "":
             break
+        
+        try:
+            read = json.loads(read)
+        except json.JSONDecodeError:
+            # Can happen with some error messages
+            job.internal_data["messages"] = job.internal_data["messages"][-4:] + [read]
+            await job.logs_fd_write((json.dumps(read) + "\n").encode("utf-8", "ignore"))
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/cloud_backup/restic.py:95]: Why `json.dumps` here?

**Comment by themylogin** [comment]
Also I don’t see any changes related to your side note, did you remove them? If so, please update the note text, as it’ll be a part of commit message when the PR will be merged, and that’ might be confusing.

**Comment by creatorcary** [comment]
@themylogin I discovered this error while trying to restore a backed-up ZFS snapshot and wanted to document it somewhere. I can make a ticket for it.

File: src/middlewared/middlewared/plugins/cloud_backup/restic.py (hunk 1 above)
Comment (creatorcary) [reply src/middlewared/middlewared/plugins/cloud_backup/restic.py:95]: You're right I'll simplify this.

**Comment by themylogin** [comment]
@creatorcary we exclude `.zfs` from both cloud sync push and pull in case `snapdev` is set to `visible`. We should probably do the same for cloud backup.

**Comment by creatorcary** [comment]
https://ixsystems.atlassian.net/browse/NAS-132608

**Review by themylogin — APPROVED** [review]

**Pull request merged**